torchvision/prototype/transforms/_auto_augment.py

```diff
@@ -0,0 +1,296 @@
+        ),
+        "TranslateX": AutoAugmentDispatcher(
+            F.affine,
+            magnitude_fn=lambda magnitude: dict(translate=[int(magnitude), 0]),
+            extra_kwargs=dict(angle=0.0, scale=1.0, shear=[0.0, 0.0]),
+            takes_interpolation_kwargs=True,
+        ),
+        "TranslateY": AutoAugmentDispatcher(
+            F.affine,
+            magnitude_fn=lambda magnitude: dict(translate=[0, int(magnitude)]),
+            extra_kwargs=dict(angle=0.0, scale=1.0, shear=[0.0, 0.0]),
+            takes_interpolation_kwargs=True,
+        ),
+        "Rotate": AutoAugmentDispatcher(F.rotate, magnitude_fn=lambda magnitude: dict(angle=magnitude)),
+        "Brightness": AutoAugmentDispatcher(
+            F.adjust_brightness, magnitude_fn=lambda magnitude: dict(brightness_factor=1.0 + magnitude)
+        ),
+        "Color": AutoAugmentDispatcher(
+            F.adjust_saturation, magnitude_fn=lambda magnitude: dict(saturation_factor=1.0 + magnitude)
+        ),
+        "Contrast": AutoAugmentDispatcher(
+            F.adjust_contrast, magnitude_fn=lambda magnitude: dict(contrast_factor=1.0 + magnitude)
+        ),
+        "Sharpness": AutoAugmentDispatcher(
+            F.adjust_sharpness, magnitude_fn=lambda magnitude: dict(sharpness_factor=1.0 + magnitude)
+        ),
+        "Posterize": AutoAugmentDispatcher(F.posterize, magnitude_fn=lambda magnitude: dict(bits=int(magnitude))),
+        "Solarize": AutoAugmentDispatcher(F.solarize, magnitude_fn=lambda magnitude: dict(threshold=magnitude)),
+        "AutoContrast": AutoAugmentDispatcher(F.autocontrast),
+        "Equalize": AutoAugmentDispatcher(F.equalize),
+        "Invert": AutoAugmentDispatcher(F.invert),
+    }
+
+    def get_transforms_meta(self, image_size: Tuple[int, int]) -> Iterator[Tuple[str, float]]:
+        raise NotImplementedError
+
+    def get_params(self, sample: Any) -> Dict[str, Any]:
+        image = Query(sample).image_for_size_and_channels_extraction()
+
+        fill = self.fill
+        if isinstance(fill, (int, float)):
+            fill = [float(fill)] * image.num_channels
+        elif fill is not None:
+            fill = [float(f) for f in fill]
+
+        return dict(interpolation=self.interpolation, fill=fill)
+
+    def forward(self, *inputs: Any, params: Optional[Dict[str, Any]] = None) -> Any:
+        sample = inputs if len(inputs) > 1 else inputs[0]
+        params = params or self.get_params(sample)
+
+        for transform_id, magnitude in self.get_transforms_meta(Query(sample).image_size()):
+            dispatcher = self._DISPATCHER_MAP[transform_id]
+
+            def transform(input: Any) -> Any:
+                if input not in dispatcher:
+                    return input
+
+                return dispatcher(input, magnitude=magnitude, **params)  # type: ignore[arg-type]
+
+            sample = apply_recursively(transform, sample)
+
+        return sample
+
+    def _randbool(self, p: float = 0.5) -> bool:
```

Comment:
This doesn't actually save us any lines of code, just adds one more method on the base class. Consider just doing `torch.rand() <= p` where necessary.